src/datasets.jl

```diff
@@ -70,3 +70,23 @@ function load_poly()
     y = convert(Vector{Float64}, raw_csv[:, 2])
     x, y, ["x", "2.6 x² + .8 x + ɛ"]
 end
+
+
+"""
+x_one, y_one, x_zero, y_zero, vars = load_spiral()`
+
+Loads an artificial example dataset for a noisy spiral function.
+It is particularly useful to explain representation learning and nonlinearity.
+The vector `x` contains 194 points between 0 and 6.5 with two classes "ones", "zeros".
+The vector `y` contains `r*sin(θ), r*cos(θ)` plus some gaussian noise
+The optional vector `vars` contains descriptive names for `x` and `y`
+"""
+function load_spiral()
+    path = joinpath(Pkg.dir("MLDataUtils"), "data", "spiral.csv")
+    raw_csv = readcsv(path)
+    x_one = convert(Vector{Float64}, raw_csv[:, 1])
+    y_one = convert(Vector{Float64}, raw_csv[:, 2])
+    x_zero = convert(Vector{Float64}, raw_csv[:, 3])
+    y_zero = convert(Vector{Float64}, raw_csv[:, 4])
+    x_one, y_one, x_zero, y_zero, ["r*sin(θ)", "r*cos(θ)", "-r*sin(θ)", "-r*cos(θ)"]
```

Comment:
We should probably provide this data-set in matrix `X` + vector `Y` form, similar to iris. So the first n points have have `Y[i] == 0` and the other half `Y[i] ==1`. The matrix would be of size `(2,194)`. This would be the typical convention to encode a numerical dataset and we should probably be consistent. That said I don't mind leaving the CSV itself as is.